openstack/compute/v2/extensions/adminactions/requests.go

```
@@ -0,0 +1,134 @@
+package adminactions
+
+import "github.com/rackspace/gophercloud"
+
+func actionURL(client *gophercloud.ServiceClient, id string) string {
+	return client.ServiceURL("servers", id, "action")
+}
+
+type CreateBackupOpts struct {
+	// Name: required, name of the backup.
+	Name string
+
+	// BackupType: required, type of the backup, such as "daily".
+	BackupType string
+
+	// Rotation: the number of backups to retain.
+	Rotation int
+}
+
+// ToBackupCreateMap assembles a request body based on the contents of a CreateOpts.
+func (opts CreateBackupOpts) ToCreateBackupMap() (map[string]interface{}, error) {
+	backup := make(map[string]interface{})
+
+	if opts.Name != "" {
+		backup["name"] = opts.Name
```

If `Name` is required, this should return an error if `opts.Name == ""`.